#! /bin/sh /usr/share/dpatch/dpatch-run
## 79_archiver_slash.dpatch by  <lionel@mamane.lu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Debian bug #350388
## DP: Some non-pipermail archivers require _no_ final slash.

@DPATCH@
diff -urNad mailman-2.1.7~/Mailman/Archiver/Archiver.py mailman-2.1.7/Mailman/Archiver/Archiver.py
--- mailman-2.1.7~/Mailman/Archiver/Archiver.py	2005-08-27 03:40:15.000000000 +0200
+++ mailman-2.1.7/Mailman/Archiver/Archiver.py	2006-02-19 10:29:18.006366380 +0100
@@ -146,8 +146,6 @@
                 'listname': self.internal_name(),
                 'hostname': hostname
                 }
-            if not url.endswith('/'):
-                url += '/'
             return url
 
     def __archive_file(self, afn):
diff -urNad mailman-2.1.7~/Mailman/Defaults.py.in mailman-2.1.7/Mailman/Defaults.py.in
--- mailman-2.1.7~/Mailman/Defaults.py.in	2005-12-30 19:50:07.000000000 +0100
+++ mailman-2.1.7/Mailman/Defaults.py.in	2006-02-19 10:29:18.007366239 +0100
@@ -205,7 +205,7 @@
 # interpolate the name of the list into this.  You can also include a
 # "%(hostname)s" in the string, into which Mailman will interpolate
 # the host name (usually DEFAULT_URL_HOST).
-PUBLIC_ARCHIVE_URL = 'http://%(hostname)s/pipermail/%(listname)s'
+PUBLIC_ARCHIVE_URL = 'http://%(hostname)s/pipermail/%(listname)s/'
 
 # Are archives on or off by default?
 DEFAULT_ARCHIVE = On
